Docs: Update docker image run and configuration instructions (#21705)
grafana/grafana: docs/sources/installation/docker.md

+++
title = "Run Grafana Docker image"
description = "Guide for running Grafana using Docker"
keywords = ["grafana", "configuration", "documentation", "docker"]
type = "docs"
[menu.docs]
name = "Run Grafana Docker image"
identifier = "docker"
parent = "installation"
weight = 600
+++

> # Run Grafana Docker image

Author: No, just missed it. Nice catch.

`grafana/grafana:<version>-ubuntu`

This image is based on [Ubuntu](https://ubuntu.com/), available in [the Ubuntu official image](https://hub.docker.com/_/ubuntu). This is an alternative image for those who prefer an [Ubuntu](https://ubuntu.com/) based image and/or who are dependent on certain tooling not available for Alpine.

> **Note:** The `grafana/grafana:<version>-ubuntu` image is available for Grafana version 6.5.0 and later.

## Run Grafana
You can run the latest Grafana version, run a specific version, or run an unstable version based on the master branch of the [grafana/grafana GitHub repository](https://github.com/grafana/grafana).

### Run the latest stable version of Grafana

> **Note:** If you are on a Linux system, you might need to add `sudo` before the command.

```bash
docker run -d -p 3000:3000 grafana/grafana
```

### Run a specific version of Grafana

> **Note:** If you are on a Linux system, you might need to add `sudo` before the command.

```bash
docker run -d -p 3000:3000 --name grafana grafana/grafana:<version number>
```

**Example:**
```bash
docker run -d -p 3000:3000 --name grafana grafana/grafana:6.5.0
```

### Run the Grafana master branch

> For every successful build of the master branch, we update the `grafana/grafana:master` and `grafana/grafana:master-ubuntu`. Additionally, two new tags are created, `grafana/grafana-dev:master-<commit hash>` and `grafana/grafana-dev:master-<commit hash>-ubuntu`, which includes the hash of the Git commit that was built. This means you can always get the latest version of Grafana.

Reviewer: ```suggestion
For every successful build of the master branch, we update the `grafana/grafana:master` and `grafana/grafana:master-ubuntu` tags. Additionally, two new tags are created, `grafana/grafana-dev:master-<commit hash>` and `grafana/grafana-dev:master-<commit hash>-ubuntu`, which includes the hash of the Git commit that was built. Use these to get access to the latest master builds of Grafana.
```

When running Grafana master in production, we *strongly* recommend that you use the `grafana/grafana-dev:master-<commit hash>` tag. This tag guarantees that you use a specific version of Grafana instead of whatever was the most recent commit at the time.

For a list of available tags, check out [grafana/grafana](https://hub.docker.com/r/grafana/grafana/tags/) and [grafana/grafana-dev](https://hub.docker.com/r/grafana/grafana-dev/tags/).

## Install plugins in the Docker container

> You can install officially-release Grafana plugins or you can install plugins from a custom URL.

Reviewer: ```suggestion
You can install official and community plugins listed on the Grafana [plugins page](https://grafana.com/grafana/plugins) or from a custom URL.
```

> ### Install official Grafana plugins

Reviewer: ```suggestion
### Install official and community Grafana plugins
```

    user: "104"
```

#### Modify permissions

The commands below runs bash inside the Grafana container with your volume mapped in. This makes it possible to modify the file ownership to match the new container. Always be careful when modifying permissions.

```bash
$ docker run -ti --user root --volume "<your volume mapping here>" --entrypoint bash grafana/grafana:5.1.0

# in the container you just started:
chown -R root:root /etc/grafana && \
chmod -R a+r /etc/grafana && \
chown -R grafana:grafana /var/lib/grafana && \
chown -R grafana:grafana /usr/share/grafana
```

### Migration from a previous version of the Docker container to 6.4 or later

Grafana Docker image was changed to be based on [Alpine](http://alpinelinux.org) instead of [Ubuntu](https://ubuntu.com/).

> ### Migration from a previous version of the docker container to 6.5 or later

Reviewer: I think we should change the ordering here to reverse chronological for the migration instructions. The latest migration instruction is most likely to be relevant to the most users.

Author: I'm game. It just means that we need to apply this change everywhere else to be consistent.
